pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -380,14 +387,40 @@ func (m *kubeGenericRuntimeManager) computePodContainerChanges(pod *api.Pod, pod
 
 	sandboxChanged, attempt, sandboxID := m.podSandboxChanged(pod, podStatus)
 	changes := podContainerSpecChanges{
-		CreateSandbox:     sandboxChanged,
-		SandboxID:         sandboxID,
-		Attempt:           attempt,
-		ContainersToStart: make(map[int]string),
-		ContainersToKeep:  make(map[kubecontainer.ContainerID]int),
-		ContainersToKill:  make(map[kubecontainer.ContainerID]containerToKillInfo),
+		CreateSandbox:        sandboxChanged,
+		SandboxID:            sandboxID,
+		Attempt:              attempt,
+		ContainersToStart:    make(map[int]string),
+		ContainersToKeep:     make(map[kubecontainer.ContainerID]int),
+		InitContainersToKeep: make(map[kubecontainer.ContainerID]int),
+		ContainersToKill:     make(map[kubecontainer.ContainerID]containerToKillInfo),
+	}
+
+	// check the status of init containers.
+	initFailed := false
+	// always reset the init containers if the sandbox is changed.
+	if !sandboxChanged {
+	Containers:
+		// keep all successfully completed containers up to and including the first failing container
+		for i, container := range pod.Spec.InitContainers {
+			containerStatus := podStatus.FindContainerStatusByName(container.Name)
```

Comment:
nilcheck and continue here, again lets the switch be specifically on containerStatus.State